spec/definitions.yaml

```diff
@@ -1019,6 +1019,24 @@ definitions:
             type: string
             description: |
               Azure VM size to use for all nodes in the node pool. _(Validated against available VM sizes.)_
+          spot_instances:
+            description: |
+              Attributes defining the presence and configuration of Spot Instances.
+            type: object
+            properties:
+              enabled:
+                description: |
+                  Whether the feature is enabled or not.
+                type: boolean
+                default: false
+              max_price:
+                description: |
+                  The maximum price that a single node pool VM instance can reach before it is deallocated.
+                  By setting this value to `-1`, the maximum price will be fixed to the on-demand price of
+                  the instance.
+                  The value can be a floating point number with a maximum of 5 decimal places.
+                type: number
+                default: -1
```

Comment:
Using a negative one int as a default appears overly complicated and counter-intuitive to me.

I would propose to interpret both an absent `max_price` attribute and `max_price: 0` as the "use maximum" instead.